test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/SolutionExplorer/Models/AssetsFileDependenciesSnapshotTests.cs

```diff
@@ -57,5 +63,82 @@ public void ParseLibraries_IgnoreCaseInDependenciesTree_Succeeds()
             Assert.Equal(1, dependencies.Count);
             Assert.True(dependencies.ContainsKey("system.runtime"));
         }
+
+        [Fact]
+        public void ParseLibraries_LockFileTargetLibrariesWithDifferentCase_Throws()
+        {
+            // Arrange
+            LockFileTarget lockFileTarget = new LockFileTarget();
+            lockFileTarget.Libraries = new List<LockFileTargetLibrary>
+            {
+                new LockFileTargetLibrary()
+                {
+                    Name = "packageA",
+                    Type = "package",
+                    Version = NuGetVersion.Parse("1.0.0")
+                },
+                new LockFileTargetLibrary()
+                {
+                    Name = "PackageA",
+                    Type = "package",
+                    Version = NuGetVersion.Parse("1.0.0")
+                }
+            };
+
+            var exception = Assert.Throws<ArgumentException>(() => AssetsFileDependenciesSnapshot.ParseLibraries(lockFileTarget));
+
+            Assert.Equal("An element with the same key but a different value already exists. Key: PackageA", exception.Message);
```

Comment:
Assertions based on strings always make me a bit weary. Are you confident this should be a string-equal check? 
I think another team owns the error, right?
I'd either rather stop short of the first Assert.Throws, or perhaps keep the second Assert, but just look for it to `.Contains("PackageA")`